src/components/robot-item-form.tsx

```diff
@@ -0,0 +1,157 @@
+import { makeStyles, TextField, Button } from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import React, { useState, useEffect } from 'react';
+import { successMsg } from '../util/alerts';
+import fakePlaces from '../mock/data/places';
+
+interface robotLoopFormProps {
+  fleetName: string;
+  requestLoop(
+    fleetName: string,
+    numLoops: number,
+    startLocationPoint: string,
+    endLocationPoint: string,
+  ): void;
+}
+
+export const RobotLoopForm = (props: robotLoopFormProps) => {
+  const { requestLoop, fleetName } = props;
+  const classes = useStyles();
+
+  const [listOfPlacesToGo, setListOfPlacesToGo] = useState(['']);
+
+  const [numLoops, setNumLoops] = useState(0);
+  const [startLocation, setStartLocation] = useState('');
+  const [finishLocation, setFinishLocation] = useState('');
+  // Error states
+  const [numLoopsError, setNumLoopsError] = useState('');
+  const [startLocationError, setStartLocationError] = useState('');
+  const [finishLocationError, setFinishLocationError] = useState('');
+
+  useEffect(() => {
```

Comment:
Instead of using an effect, it might be better to use lazy state initialization or receiving the list of places as a prop to avoid double rendering on the initial mount.

Reply:
Makes sense, I'll pass it as a prop. We will deal with that list in the future again when we can get the data from the backend. Thanks!